kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerLagManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import io.confluent.kafkarest.entities.ConsumerLag;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A service to manage Kafka {@link ConsumerLag Consumer Lags}.
+ */
+public interface ConsumerLagManager {
+
+  /**
+   * Returns the list of Kafka {@link ConsumerLag Consumer Lags} belonging to the
+   * {@link io.confluent.kafkarest.entities.ConsumerGroup} with the given {@code consumerGroupId}.
+   */
+  CompletableFuture<List<ConsumerLag>> listConsumerLags(
+      String clusterId, String consumerGroupId);
+
+  /**
+   * Returns the Kafka {@link ConsumerLag} with the given {@code consumerId}.
+   */
+  CompletableFuture<Optional<ConsumerLag>> getConsumerLag(
+      String clusterId, String topicName, Integer partitionId, String consumerGroupId);
```

Comment:
This ordering of parameters does not make sense, does it?

It should be `clusterId, consumerGroupId, topicName, partitionId`. 

If you have listFoo(x, y), and getFoo, getFoo is a refinement of listFoo, and the parameters should be strictly an extension of the parameters of listFoo, e.g. x, y, z.

Reply:
I took the ordering from a previous conversation - https://github.com/confluentinc/kafka-rest/pull/765#discussion_r539581651 (for 'Specific lag for given topic-partition/consumer-group'). Should I change the order of parameters everywhere?

Reply:
It should be either (1) `clusterId, topicName, partitionId, consumerGroupId, consumerId...` or (2) `clusterId, consumerGroupId, consumerId..., topicName, partitionId`. Pick one, doesn't matter which one, and use it everywhere. My comment here is that listConsumerLags is using `clusterId, consumerGroupId`, which means you probably chose (2), so you should use (2) everywhere.

Reply:
@rigelbm just to confirm, if I use 
`/clusters/{cluster_id}/consumer-group/{consumer_group_id}/lags` 
for listing consumer lag, 
then I should use (something along the lines of) 
`/clusters/{cluster_id}/consumer-group/{consumer_group_id}/lags/topics/{topic_name}/partitions/{partition_id}` 
or 
`/clusters/{cluster_id}/consumer-group/{consumer_group_id}/lags/{topic_name}/partitions/{partition_id}` 
for individual consumer lags?

Reply:
`/clusters/{cluster_id}/consumer-group/{consumer_group_id}/topics/{topic_name}/partitions/{partition_id}/lag`
would also work? though I think this means I would need to split the Get and List into two separate Resource files (like how they are currently)